azurerm/helpers/azure/batch_pool.go

```diff
@@ -123,6 +123,20 @@ func FlattenBatchPoolStartTask(startTask *batch.StartTask) []interface{} {
 		result["user_identity"] = []interface{}{userIdentity}
 	}
 
+	resourceFiles := make([]interface{}, 0)
+	if startTask.ResourceFiles != nil {
+		for _, armResourceFile := range *startTask.ResourceFiles {
+			resourceFile := make(map[string]interface{})
+			resourceFile["auto_storage_container_name"] = armResourceFile.AutoStorageContainerName
+			resourceFile["storage_container_url"] = armResourceFile.StorageContainerURL
+			resourceFile["http_url"] = armResourceFile.HTTPURL
+			resourceFile["blob_prefix"] = armResourceFile.BlobPrefix
+			resourceFile["file_path"] = armResourceFile.FilePath
+			resourceFile["file_mode"] = armResourceFile.FileMode
```

Comment:
since these are [all pointers](https://github.com/Azure/azure-sdk-for-go/blob/master/services/batch/mgmt/2018-12-01/batch/models.go#L2650-L2661) - can we nil-check around them and then de-reference them (to ensure they're set correctly); for example:

```
if armResourceFile.AutoStorageContainerName != nil {
  resourceFile["auto_storage_container_name"] = *armResourceFile.AutoStorageContainerName
}
if armResourceFile.StorageContainerURL != nil {
  resourceFile["storage_container_url"] = *armResourceFile.StorageContainerURL
}
if armResourceFile.HTTPURL != nil {
  resourceFile["http_url"] = *armResourceFile.HTTPURL
}
if armResourceFile.BlobPrefix != nil {
  resourceFile["blob_prefix"] = *armResourceFile.BlobPrefix
}
if armResourceFile.FilePath != nil {
  resourceFile["file_path"] = *armResourceFile.FilePath
}
if armResourceFile.FileMode != nil {
  resourceFile["file_mode"] = *armResourceFile.FileMode
}
```